core/src/cats/sql_create.cc

```diff
@@ -102,21 +102,25 @@ bool BareosDb::CreateJobRecord(JobControlRecord* jcr, JobDbRecord* jr)
  */
 bool BareosDb::CreateJobmediaRecord(JobControlRecord* jcr, JobMediaDbRecord* jm)
 {
-  int count;
-  char ed1[50], ed2[50], ed3[50];
-
   DbLocker _{this};
 
   /* clang-format off */
   Mmsg(cmd,
-       "UPDATE JobMedia SET LastIndex=%lu, EndFile=%lu, EndBlock=%lu, JobBytes=%s "
-       "WHERE JobId=%s AND MediaId=%s",
+       "UPDATE JobMedia SET "
+       "FirstIndex = (CASE when firstindex=0 then %lu else firstindex end), "
+       "StartBlock = (CASE when startblock=0 then %lu else startblock end), "
+       "StartFile  = (CASE when startfile=0 then %lu else startfile end), "
```

Comment:
Are we sure that `firstindex`, `startblock` and `startfile` are always > 0 in normal operation, and only the dummy place holder sets them to 0?

Reply:
only the dummy placeholder creates zero-filled jobmedia record, no other place in the code requires those fields to be zero.

Reply:
as we're overwriting placeholder data with real data, I would consider this safe (even though it isn't nice).